### PR TITLE
Ensure that partially assigned structures are properly hoisted by async rewriter.

### LINKED ISSUE #1942: Async rewriter does not lift partially initialized structs in /o+ mode.
```C#
using System;
using System.Threading.Tasks;
namespace ConsoleApp
{
    struct TestStruct
    {
        public long i;
        public long j;
    }
    class Program
    {
        static async Task TestAsync()
        {
            TestStruct t;
            t.i = 12;
            Console.WriteLine("Before {0}", t.i); // emits "Before 12" 
            await Task.Delay(100);
            Console.WriteLine("After {0}", t.i); // emits "After 0" expecting "After 12" 
        }
        static void Main(string[] args)
        {
            TestAsync().Wait();
        }
    }
}
```

=== in Debug prints  (correctly):
Before 12
After 12

=== in Release prints 
Before 12
After 0

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #1942.

### CODE REVIEW

**Comment by gafter** [comment]
LGTM

**Comment by AlekseyTs** [comment]
@VSadov, @jaredpar, @agocke Please review.

File: src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/IteratorAndAsyncCaptureWalker.cs [hunk 1]
```diff
@@ -130,17 +130,33 @@ private void MarkLocalsUnassigned()
...
+
+                        case SymbolKind.Parameter:
+                            SetSlotState(i, false);
+                            break;
+
+                        case SymbolKind.Field:
+                            if (!((FieldSymbol)symbol).IsConst)
+                            {
+                                SetSlotState(i, false);
+                            }
```
Comment (VSadov) [inline src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/IteratorAndAsyncCaptureWalker.cs]: Do we care whether it is a static/instance field or whether container is a class/struct?

Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/IteratorAndAsyncCaptureWalker.cs]: I believe that only the "right kind" fields will be assigned a slot.

**Comment by VSadov** [comment]
LGTM

**Pull request merged**